packages/file-store/configstores/FileConfigstore.ts

```diff
@@ -37,13 +37,13 @@ export class FileConfigstore implements Configstore {
 
   async list(): Promise<Array<string>> {
     return this.queue.add(async () => {
-      const files = await fs.readdir(this.directory, {withFileTypes: true})
-      const promises = files
-        .filter((file) => file.isFile() && file.name.endsWith('.json'))
-        .map((file) => fs.readFile(path.resolve(file.path, file.name), 'utf8'))
-
-      return Promise.all(promises)
-    }) as Promise<string[]>
+      const files = await fs.readdir(this.directory)
+      // readdir returns files without file extension.
+      // This means we will get two files,
+      // the actual file and the JSON info file, per upload.
+      // We de-duplicate here to only return one ID per upload.
+      return Array.from(new Set(files))
```

Comment:
I have tested this
`fs.readdir` still return files with extension
![image](https://github.com/tus/tus-node-server/assets/104412855/d976f27c-a82c-48f7-b5d9-0520e2e2481a)

Reply:
Thanks for double checking! Not sure how I missed this.